x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/custom_timeline_data_grid_body.tsx

```diff
@@ -42,31 +54,137 @@ const DEFAULT_UDT_ROW_HEIGHT = 34;
  * */
 export const CustomTimelineDataGridBody: FC<CustomTimelineDataGridBodyProps> = memo(
   function CustomTimelineDataGridBody(props) {
-    const { Cell, visibleColumns, visibleRowData, rows, rowHeight, enabledRowRenderers, refetch } =
-      props;
+    const {
+      Cell,
+      visibleColumns,
+      visibleRowData,
+      rows,
+      rowHeight,
+      enabledRowRenderers,
+      refetch,
+      setCustomGridBodyProps,
+      headerRow,
+      footerRow,
+      gridWidth,
+    } = props;
+
+    const { euiTheme } = useEuiTheme();
+
+    // // Set custom props onto the grid body wrapper
+    const bodyRef = useRef<HTMLDivElement | null>(null);
+    useEffect(() => {
+      setCustomGridBodyProps({
+        ref: bodyRef,
+        style: {
+          width: '100%',
+          height: '100%',
+          overflowY: 'hidden',
+          scrollbarColor: `${euiTheme.colors.mediumShade} ${euiTheme.colors.lightestShade}`,
+        },
+      });
+    }, [setCustomGridBodyProps, euiTheme.colors.mediumShade, euiTheme.colors.lightestShade]);
 
     const visibleRows = useMemo(
       () => (rows ?? []).slice(visibleRowData.startRow, visibleRowData.endRow),
       [rows, visibleRowData]
     );
 
+    const listRef = useRef<VariableSizeList<unknown>>(null);
+
+    const rowHeights = useRef<number[]>([]);
+
+    const setRowHeight = useCallback((index: number, height: number) => {
+      if (rowHeights.current[index] === height) return;
+      listRef.current?.resetAfterIndex(index);
+
+      rowHeights.current[index] = height;
+    }, []);
+
+    const getRowHeight = useCallback((index: number) => {
+      return rowHeights.current[index] ?? 100;
+    }, []);
+
+    const innerRowContainer = useMemo(() => {
+      const InnerComp = React.forwardRef<HTMLDivElement, PropsWithChildren<{}>>(
```

Comment:
can probably pull this definition to the top level of this file, and that way this is never recreated.

Reply:
yes.. I was just waiting for the EUI teams opinion and then i will refactor code to be put in production. Currently, this is just working code. 

I wanted to get everyone opinions mainly on performance. But what you say makes sense.

Reply:
Understood, I like the changes, ultimately though I think to finally get the performance to where we really want, we will need to continue to refactor on the eui side. Specifically, useRowHeightUtils and all the associated hooks can likely be cleaned up, especially for use in data_grid_body_custom.tsx, where virtualization is not used at all.

Reply:
@kqualters-elastic , I just checked, it has depenendencies `headerRow` and `footerRow` and hence it cannot be pulled out.